tokio/src/lib.rs

```diff
@@ -487,6 +487,23 @@ compile_error!("Tokio's build script has incorrectly detected wasm.");
 ))]
 compile_error!("Only features sync,macros,io-util,rt,time are supported on wasm.");
 
+#[cfg(all(not(tokio_unstable), tokio_taskdump))]
+compile_error!("The `tokio_taskdump` feature requires `--cfg tokio_unstable`.");
+
+#[cfg(all(
+    tokio_unstable,
+    tokio_taskdump,
+    feature = "rt",
```

Comment:
I would trigger this compile error even without `rt`. You aren't going to be enabling `--cfg tokio_taskdump` without knowing about it. Possibly also compile-fail on `--cfg tokio_taskdump` without rt.

Reply:
> Possibly also compile-fail on `--cfg tokio_taskdump` without rt.

Hm, unfortunately, this causes both a [CI failure](https://github.com/tokio-rs/tokio/actions/runs/4745864579/jobs/8428808277?pr=5608#step:7:81) an a usability issue: You might just keep `--cfg tokio_unstable --cfg tokio_taskdump` in your environment (.cargo/config makes this easy to do), but *not* always want to actually use the taskdumping feature. In such scenarios, your build will fail because of the explicit `compile_error!`.